### ISSUE TITLE
Error when loading a save slot other than the one saved in

### ISSUE THREAD

**Issue by itsschwer** [chameleonkid/Secrets#58]
Appears to be related to loading the texture for the player's appearance.

Error stack trace:
```
NullReferenceException: Object reference not set to an instance of an object
ES3Types.ES3Type_Texture2D.ReadUnityObject[T] (ES3Reader reader, System.Object obj) (at Assets/Plugins/Easy Save 3/Scripts/Types/Unity Types/ES3Type_Texture2D.cs:37)
ES3Types.ES3UnityObjectType.ReadObject[T] (ES3Reader reader) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3UnityObjectType.cs:98)
ES3Types.ES3ObjectType.Read[T] (ES3Reader reader) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3ObjectType.cs:54)
ES3Reader.ReadObject[T] (ES3Types.ES3Type type) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:249)
ES3Reader.Read[T] (ES3Types.ES3Type type) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:278)
ES3Types.ES3Type.ReadProperties (ES3Reader reader, System.Object obj) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3Type.cs:154)
ES3Types.ES3ReflectedScriptableObjectType.ReadScriptableObject[T] (ES3Reader reader, System.Object obj) (at Assets/Plugins/Easy Save 3/Scripts/Types/Reflected Types/ES3ReflectedScriptableObjectType.cs:24)
ES3Types.ES3ScriptableObjectType.ReadObject[T] (ES3Reader reader) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3ScriptableObjectType.cs:69)
ES3Types.ES3ObjectType.Read[T] (ES3Reader reader) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3ObjectType.cs:54)
ES3Reader.ReadObject[T] (ES3Types.ES3Type type) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:249)
ES3Reader.Read[T] (ES3Types.ES3Type type) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:278)
ES3Reader.Read[T] (System.String key, T defaultValue) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:212)
ES3.Load[T] (System.String key, T defaultValue, ES3Settings settings) (at Assets/Plugins/Easy Save 3/Scripts/ES3.cs:419)
ES3.Load[T] (System.String key, System.String filePath, T defaultValue) (at Assets/Plugins/Easy Save 3/Scripts/ES3.cs:391)
SimpleSave.LoadAppearance (System.String loadSlot) (at Assets/Scripts/GameManagers/SaveSystem/SimpleSave.cs:70)
SimpleSave.Load (System.String loadSlot) (at Assets/Scripts/GameManagers/SaveSystem/SimpleSave.cs:36)
PauseManager.Load (System.String loadSlot) (at Assets/Scripts/Menus/PauseManager.cs:62)
```

**Comment by chameleonkid** [comment]
I cant reproduce this.

**Comment by itsschwer** [comment]
To reproduce:
1. Start a game from the scene `CharacterCreation`
2. Save the character to load `Mavens_Inn_Cutscene`
3. Use the SavePoint in `Mavens_Inn_Cutscene`, saving to slot 1
4. Exit Play Mode
5. Repeat steps 1 & 2.
6. Attempt to load slot 1 from the pause menu

**Comment by itsschwer** [comment]
The previous error stack trace was on a development branch. Here's the error stack trace from the `main` branch:
```
NullReferenceException: Object reference not set to an instance of an object
ES3Types.ES3Type_Texture2D.ReadUnityObject[T] (ES3Reader reader, System.Object obj) (at Assets/Plugins/Easy Save 3/Scripts/Types/Unity Types/ES3Type_Texture2D.cs:37)
ES3Types.ES3UnityObjectType.ReadObject[T] (ES3Reader reader) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3UnityObjectType.cs:98)
ES3Types.ES3ObjectType.Read[T] (ES3Reader reader) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3ObjectType.cs:54)
ES3Reader.ReadObject[T] (ES3Types.ES3Type type) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:249)
ES3Reader.Read[T] (ES3Types.ES3Type type) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:278)
ES3Types.ES3Type.ReadProperties (ES3Reader reader, System.Object obj) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3Type.cs:154)
ES3Types.ES3ReflectedScriptableObjectType.ReadScriptableObject[T] (ES3Reader reader, System.Object obj) (at Assets/Plugins/Easy Save 3/Scripts/Types/Reflected Types/ES3ReflectedScriptableObjectType.cs:24)
ES3Types.ES3ScriptableObjectType.ReadObject[T] (ES3Reader reader) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3ScriptableObjectType.cs:69)
ES3Types.ES3ObjectType.Read[T] (ES3Reader reader) (at Assets/Plugins/Easy Save 3/Scripts/Types/ES3ObjectType.cs:54)
ES3Reader.ReadObject[T] (ES3Types.ES3Type type) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:249)
ES3Reader.Read[T] (ES3Types.ES3Type type) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:278)
ES3Reader.Read[T] (System.String key, T defaultValue) (at Assets/Plugins/Easy Save 3/Scripts/Readers/ES3Reader.cs:212)
ES3.Load[T] (System.String key, T defaultValue, ES3Settings settings) (at Assets/Plugins/Easy Save 3/Scripts/ES3.cs:419)
ES3.Load[T] (System.String key, System.String filePath, T defaultValue) (at Assets/Plugins/Easy Save 3/Scripts/ES3.cs:391)
SimpleSave.LoadAppearance (System.String loadSlot) (at Assets/Scripts/GameManagers/SaveSystem/SimpleSave.cs:68)
SimpleSave.Load (System.String loadSlot) (at Assets/Scripts/GameManagers/SaveSystem/SimpleSave.cs:33)
PauseManager.Load (System.String loadSlot) (at Assets/Scripts/Menus/PauseManager.cs:73)
```

**Comment by itsschwer** [comment]
Seems to be due to a missing `ES3ReferenceMgr` in the `CharacterCreation` scene (solved by @chameleonkid!)

**Comment by itsschwer** [comment]
Resolved by commit e6f0cec.

ES3 now gives this warning:
> Easy Save cannot load the properties or pixels for this Texture because it is not read/write enabled, so it will be loaded by reference. To load the properties and pixels for this Texture, check the 'Read/Write Enabled' checkbox in its Import Settings.
To disable warnings from Easy Save, go to Window > Easy Save 3 > Settings, and uncheck 'Log Warnings'

We *do* want to load the texture by reference, since we aren't modifying the textures themselves at runtime (just which ones should be used for the player). ES3 seems to lack options to disable specific warnings, so the only option currently is just to ignore the above error when loading saves.
